pallets/aura-ext/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+// Copyright 2021 Parity Technologies (UK) Ltd.
+// This file is part of Cumulus.
+
+// Cumulus is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Cumulus is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Cumulus.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Cumulus extension pallet for AuRa
+//!
+//! This pallets extends the Substrate AuRa pallet to make it compatible with parachains. It
+//! provides the [`Pallet`], the [`Config`] and the [`GenesisConfig`].
+//!
+//! It is also required that the parachain runtime uses the provided [`BlockExecutor`] to properly
+//! check the constructed block on the relay chain.
+//!
+//! ```
+//!# struct Runtime;
+//!# struct Executive;
+//! cumulus_pallet_parachain_system::register_validate_block!(
+//!     Runtime,
+//!     cumulus_pallet_aura_ext::BlockExecutor::<Runtime, Executive>,
+//! );
+//! ```
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::traits::{ExecuteBlock, FindAuthor};
+use sp_application_crypto::RuntimeAppPublic;
+use sp_consensus_aura::digests::CompatibleDigestItem;
+use sp_runtime::traits::{Block as BlockT, Header as HeaderT};
+
+type Aura<T> = pallet_aura::Pallet<T>;
+
+pub use pallet::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use frame_support::pallet_prelude::*;
+	use frame_system::pallet_prelude::*;
+	use sp_std::vec::Vec;
+
+	/// The configuration trait.
+	#[pallet::config]
+	pub trait Config: pallet_aura::Config + frame_system::Config {}
+
+	#[pallet::pallet]
+	pub struct Pallet<T>(_);
+
+	#[pallet::hooks]
+	impl<T: Config> Hooks<BlockNumberFor<T>> for Pallet<T> {
+		fn on_finalize(_: BlockNumberFor<T>) {
+			// Update to the latest AuRa authorities.
+			Authorities::<T>::put(Aura::<T>::authorities());
+		}
+
+		fn on_initialize(_: BlockNumberFor<T>) -> Weight {
+			// Fetch the authorities once to get them into the storage proof of the PoV.
+			Authorities::<T>::get();
+
+			T::DbWeight::get().reads_writes(2, 1)
+		}
+	}
+
+	#[pallet::call]
+	impl<T: Config> Pallet<T> {}
+
+	/// Serves as cache for the authorities.
+	///
+	/// The authorities in AuRa are overwritten in `on_initialize` when we switch to a new session,
+	/// but we require the old authorities to verify the seal when validating a PoV. This will always
+	/// be updated to the latest AuRa authorities in `on_finalize`.
+	#[pallet::storage]
+	pub(crate) type Authorities<T: Config> = StorageValue<_, Vec<T::AuthorityId>, ValueQuery>;
```

Comment:
Is it possible to introduce bounded vec here so we dont need to change it later?